### ISSUE TITLE
set accessor is inaccessible due to its protection level

### ISSUE THREAD

**Issue by damianday** [beefytech/Beef#67]
Internal modifier is not accessible, same problem when trying to use an extension to the class.

```
        public class TestClass
	{
		struct Record
		{
			public int Value { get; internal set; }
		}

		Record r;

		public void Test()
		{
			r.Value = 5;
		}
	}
```

**Comment by bfiete** [comment]
We copied the 'internal' protection from C#, but it never really fit- we have a more generalized way to pierce protection with [Friend], and protection based on just being defined in the same project inhibits cross-project refactoring and just seems wrong to me.

So "internal" is now out. You can use "r.[Friend]Value = 5;" if 'set' were private. [Friend] now works properly for private ctors and dtors too.

**Comment by damianday** [comment]
Fixed by 77e204deb0ef72308b41e1930414dddff3b2e630.
Friend accessor is now working fine in properties
